src/modules/site-v2/base/static/js/scatter-histogram.js

```diff
@@ -0,0 +1,253 @@
+/* Plot a set of data using a scatterplot with histograms along the axes.
+ *
+ * Adapted from https://gist.github.com/larsenmtl/b00fcedde4d3d37098509f514a354bac
+ * Inspired by https://seaborn.pydata.org/generated/seaborn.jointplot.html
+ */
+
+
+
+/* Helper function to compute the extent (the [min, max]) of a set of data,
+ * plus a fixed "buffer" amount on either side.
+ *
+ * Equivalent to [min - buffer, max + buffer].
+ * 
+ * Arguments:
+ *   - data (arr):   The list of data. See D3's implementation of `extent` for more details.
+ *   - buffer (int): The amount to expand the range by.
+ *   - f (function): A function to apply to each data point before using it to find the min or max.
+ *                   See D3's implementation of `extent` for more details.
+ */
+function buffered_extent(data, buffer, f=null) {
+  const r = d3.extent(data, f);
+  r[0] -= buffer;
+  r[1] += buffer;
+  return r;
+}
+
+
+
+/* Add a histogram along an axis of a graph.
+ *
+ * Arguments:
+ *   - d (int): The dimension of the histogram. 0 for x-axis, 1 for y-axis
+ *   - axis: A d3 scaleLinear object mapping the domain of the given axis to its position in the graph
+ *   - data: The data to plot
+ *   - target: The SVG element to create the histogram in
+ *   - config: Optional keyword arguments.
+ */
+function add_histogram_along_axis(d, axis, data, target, config) {
+
+  // Validate the dimension parameter
+  if (!(d === 0 || d === 1)) {
```

Comment:
Optional, since this is part of the API for the histogram, but consider upgrading `0` and `1` to enums to avoid 'magic numbers'. 
```
const HistogramDimension = {
  'x_axis': 0,
  'y_axis': 1
};
```

Reply:
Added in 7651b21e7c0ec4105403c84ddef0a8bb07ff348c